src/DataValidator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace League\ISO3166;
+
+class DataValidator
+{
+    use KeyValidators;
+
+    /**
+     * @param array $data
+     *
+     * @return array
+     */
+    public function validate(array $data)
+    {
+        foreach ($data as $entry) {
+            $this->assertEntryHasRequiredKeys($entry);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param array $entry
+     *
+     * @throws \DomainException if given data entry does not have all the required keys.
+     */
+    private function assertEntryHasRequiredKeys(array $entry)
```

Comment:
I'd opt to let it remain private, since there's currently no use-case for single entry validation. Everything that's `public` is part of the public API thus will impact versioning.